tools/apigee-x-trial-provision/apigee-x-trial-provision.sh

```diff
@@ -0,0 +1,186 @@
+export RUNTIME_IP
+
+# 3. Create a firewall rule that lets the Load Balancer access Envoy
+gcloud compute firewall-rules create k8s-allow-lb-to-apigee-envoy \
+  --description "Allow incoming from GLB on TCP port 443 to Apigee Proxy" \
+  --network $NETWORK --allow=tcp:443 \
+  --source-ranges=130.211.0.0/22,35.191.0.0/16 --target-tags=gke-apigee-proxy --project "$PROJECT"
+
+# Step 7e: Upload credentials
+# 1. Generate a certificate and key. 
+
+openssl req -x509 -out $RUNTIME_SSL_CERT -keyout $RUNTIME_SSL_KEY -newkey rsa:2048 -nodes -sha256 -subj '/CN='"$RUNTIME_HOST_ALIAS"'' -extensions EXT -config <( printf "[dn]\nCN=$RUNTIME_HOST_ALIAS\n[req]\ndistinguished_name=dn\n[EXT]\nbasicConstraints=critical,CA:TRUE,pathlen:1\nsubjectAltName=DNS:$RUNTIME_HOST_ALIAS\nkeyUsage=digitalSignature,keyCertSign\nextendedKeyUsage=serverAuth")
+
+# 2. Upload your SSL server certificate and key to GCP
+gcloud compute ssl-certificates create apigee-ssl-cert \
+  --certificate=$RUNTIME_SSL_CERT \
+  --private-key=$RUNTIME_SSL_KEY --project "$PROJECT"
+
+# Step 7f: Create a global Load Balancer
+
+# 1. Create a health check:
+gcloud compute health-checks create https hc-apigee-envoy-443 \
+  --port 443 --global \
+  --request-path /healthz/ingress --project "$PROJECT"
+
+# 2. Create a backend service called "apigee-envoy-backend"
+
+gcloud compute backend-services create apigee-envoy-backend \
+  --protocol HTTPS --health-checks hc-apigee-envoy-443 \
+  --port-name https --timeout 60s --connection-draining-timeout 300s --global --project "$PROJECT"
+
+# 3. Add the Envoy instance group to your backend service
+gcloud compute backend-services add-backend apigee-envoy-backend \
+  --instance-group "$MIG" \
+  --instance-group-region "$REGION" \
+  --balancing-mode UTILIZATION --max-utilization 0.8 --global --project "$PROJECT"
+
+# 4. Create a Load Balancing URL map
+gcloud compute url-maps create apigee-envoy-proxy-map \
+  --default-service apigee-envoy-backend --project "$PROJECT"
+
+# 5. Create a Load Balancing target HTTPS proxy 
+gcloud compute target-https-proxies create apigee-envoy-https-proxy \
+  --url-map apigee-envoy-proxy-map \
+  --ssl-certificates apigee-ssl-cert --project "$PROJECT"
+
+# 6. Create a global forwarding rule
+gcloud compute forwarding-rules create apigee-envoy-https-lb-rule \
+  --address lb-ipv4-vip-1 --global \
+  --target-https-proxy apigee-envoy-https-proxy --ports 443 --project "$PROJECT"
+
+## TODO: [ ] wait till LB is ready
+
+
+echo ""
+echo "# To send a test request, execute following commands:"
+echo ""
+echo "export RUNTIME_IP=$RUNTIME_IP"
+
+echo "export RUNTIME_SSL_CERT=~/mig-cert.pem"
+echo "export RUNTIME_SSL_KEY=~/mig-key.pem"
+echo "export RUNTIME_HOST_ALIAS=$RUNTIME_HOST_ALIAS"
+
+echo 'curl --cacert $RUNTIME_SSL_CERT https://$RUNTIME_HOST_ALIAS/hello-world -v --resolve "$RUNTIME_HOST_ALIAS:443:$RUNTIME_IP"'
```

Comment:
remind me, do we automatically deploy a hello world proxy to every X eval org?

Reply:
![image](https://user-images.githubusercontent.com/43555662/108042717-15229700-7038-11eb-8a7b-db0eb769c58e.png)
yup!